pyaedt/modules/solutions.py

```diff
@@ -1438,492 +1352,325 @@ def assign_weight(self, a, b, taper="flat", port_cont=0):
-                    self._app.logger.error(export_path + exported_name_base + ".ffd")
-
-            else:
-                self._app.logger.info("Using Existing Embedded Element Patterns")
-            local_path = "{}/{}/eep/".format(settings.remote_rpc_session_temp_folder, full_setup_str)
-            export_path = check_and_download_folder(local_path, export_path)
-            if os.path.exists(export_path + "/" + exported_name_map):
-                with open(export_path + "/" + exported_name_map, "r") as reader:
-                    lines = [line.split(None) for line in reader]
-                lines = lines[1:]  # remove header
-                for pattern in lines:
-                    if len(pattern) >= 2:
-                        port = pattern[0]
-                        if ":" in port:
-                            port = port.split(":")[0]
-                        path_dict[-1][port] = export_path + "/" + pattern[1] + ".ffd"
-        elapsed_time = time.time() - time_before
-        self._app.logger.info("Exporting Embedded Element Patterns...Done: %s seconds", elapsed_time)
-        return path_dict
-
+        farfield_data["Theta"] = theta_range
+        farfield_data["Phi"] = phi_range
+        farfield_data["nPhi"] = Nphi
+        farfield_data["nTheta"] = Ntheta
+        farfield_data["Pincident"] = incident_power
+        real_gain = 2 * np.pi * np.abs(np.power(farfield_data["rETotal"], 2)) / incident_power / 377
+        farfield_data["RealizedGain"] = real_gain
+        farfield_data["RealizedGain_Total"] = real_gain
+        farfield_data["RealizedGain_dB"] = 10 * np.log10(real_gain)
+        real_gain = 2 * np.pi * np.abs(np.power(farfield_data["rETheta"], 2)) / incident_power / 377
+        farfield_data["RealizedGain_Theta"] = real_gain
+        real_gain = 2 * np.pi * np.abs(np.power(farfield_data["rEPhi"], 2)) / incident_power / 377
+        farfield_data["RealizedGain_Phi"] = real_gain
+        farfield_data["Element_Location"] = port_positions
+        return farfield_data
+
+    # fmt: off
     @pyaedt_function_handler()
     def plot_farfield_contour(
-        self,
-        qty_str="RealizedGain",
-        phi_scan=0,
-        theta_scan=0,
-        title="RectangularPlot",
-        convert_to_db=True,
-        export_image_path=None,
+            self,
+            farfield_quantity="RealizedGain",
+            phi_scan=0,
+            theta_scan=0,
+            title="RectangularPlot",
+            format_quantity="dB10",
+            export_image_path=None,
+            levels=64,
+            show=True,
+            **kwargs
     ):
+        # fmt: on
         """Create a Contour plot of specified quantity.
 
         Parameters
         ----------
-        qty_str : str, optional
-            Quantity to plot. Default `"RealizedGain"`.
+        farfield_quantity : str, optional
+            Far field quantity to plot. The default is`"RealizedGain"`.
```

Comment:
```suggestion
            Far field quantity to plot. The default is ``"RealizedGain"``.
```
Double backticks are necessary. Otherwise, the phrase is merely italicized instead of formatted as a code entity.